docs/runtime/go-metrics.md

```diff
@@ -0,0 +1,173 @@
+  - [Metric: `go.memory.used`](#metric-gomemoryused)
+  - [Metric: `go.memory.limit`](#metric-gomemorylimit)
+  - [Metric: `go.memory.allocated`](#metric-gomemoryallocated)
+  - [Metric: `go.memory.allocations`](#metric-gomemoryallocations)
+- [Go Garbage Collection](#go-garbage-collection)
+  - [Metric: `go.memory.gc.target`](#metric-gomemorygctarget)
+  - [Metric: `go.memory.gc.user_target`](#metric-gomemorygcuser_target)
+- [Go Threads](#go-threads)
+  - [Metric: `go.thread.limit`](#metric-gothreadlimit)
+- [Go Scheduler](#go-scheduler)
+  - [Metric: `go.schedule.duration`](#metric-goscheduleduration)
+
+<!-- tocstop -->
+
+## Go Memory
+
+**Description:** Go runtime metrics captured under the namespace `go.memory.*`
+
+### Metric: `go.memory.used`
+
+This metric is [recommended][MetricRecommended].
+This metric is obtained from Go's [`runtime/metrics`][RuntimeMetrics] package using `/memory/classes` metrics.
+
+<!-- semconv metric.go.memory.used(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.used` | UpDownCounter | `By` | Virtual memory mapped by the Go runtime. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.used(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| `go.memory.type` | string | The type of memory. | `released`; `stack` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+`go.memory.type` MUST be one of the following:
+
+| Value  | Description | Stability |
+|---|---|---|
+| `released` | Memory that is completely free and has been returned to the underlying system. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `stack` | Memory allocated from the heap that is reserved for stack space, whether or not it is currently in-use. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `other` | All memory mapped by the Go runtime into the current process as read-write, excluding other categories of memory usage. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+### Metric: `go.memory.limit`
+
+This metric is [recommended][MetricRecommended].
+This metric is obtained from Go's [`runtime/metrics`][RuntimeMetrics] package using `/gc/gomemlimit:bytes`.
+
+<!-- semconv metric.go.memory.limit(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.limit` | UpDownCounter | `By` | Go runtime memory limit configured by the user, otherwise math.MaxInt64. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
How is this memory limit retrieved? `GOMEMLIMIT`?

`math.MaxInt64` seems a bit unhelpful if someone hasn't explicitly set a limit.
Could this be the available memory? (it may cause performance issues to retrieve it though)

Reply:
> How is this memory limit retrieved? GOMEMLIMIT?

I think via a `runtime/metrics.Read()` call for `/gc/gomemlimit:bytes`.

Reply:
Right. My concern was more around the use of `math.MaxInt64` as the default limit, which doesn't provide much value. Could we use `/proc/meminfo` to retrieve the total amount of memory instead?

Reply:
I would prefer not recording a metric at all, if there is no limit imposed by the go runtime than trying to discover an external memory limit.  I've updated the PR to exclude max int-valued metrics.